src/drivers/px4fmu/fmu.cpp

```diff
@@ -1556,7 +1556,7 @@ PX4FMU::cycle()
 			if (_rc_scan_begin == 0) {
 				_rc_scan_begin = _cycle_timestamp;
 				// Configure serial port for SBUS
-				sbus_config(_rcs_fd, false);
+				sbus_config(_rcs_fd, board_supports_single_wire());
```

Comment:
- Does single wire apply only to SBUS or to other protocols as well?
- `board_supports_single_wire` is a bit confusing since single wire is also used at other places. Can you rename that to be more specific, like `board_use_single_wire_rc`?

Reply:
@bkueng It is indeed board-support single wire, not just RC. It depends on the MCU. And it depends on various protocols, but you need to look at it from the board and MCU perspective, so configuring it this way is the right way.

Reply:
Ok, if it's protocol dependent, then the other protocols need to disable it when switching different scanning modes.

Reply:
@bkueng, @LorenzMeier 

> since single wire is also used at other places' 

@bkueng I was not aware of this. What is the use case you are referring to. 

This is really a wiring issue on a given board. Maybe I am limiting it to much.

In this case it is wired one way. Meaning the connector that is connected to this port is wired only one way.  Only the SoC's TX pin is wired to the connector. 

c
p
u Non one wire  | connector  
|------------------> Tx 
|
|------------------< Rx

c
p
u one wire    | (connector)
|---------------> Tx 
|
|------------rx Not connected
 Here Become an input unless the UART is transmitting

From a HW perspective this signal will have a contention on it (TX driving a TX) if the UxART is not placed in single wire mode. So the change is made once and not undone as all the protocols need to RX on this pin.

![image](https://user-images.githubusercontent.com/1945821/42685763-bcfa7aba-8648-11e8-9995-80e00076b874.png)

So in a way logically

`board_has_single_wire_rc =  sbus_config(_rcs_fd, board_supports_single_wire());`

Is true: When board_supports_single_wire() is true and  associated with a tty (UxART).
sbus_config makes the association.

The setting could be made in the board code and not exposed, but i believe in the future the protocol needs to be configured for half duplex (with arbitration). 

So one could argue 'board_supports_single_wire(device name)' may make more sense to generalize the API

What are both your thoughts?